[1.7.4?] Add LONG_WEAPON bonus (#7204)
vcmi/vcmi: docs/modders/Bonus/Bonus_Types.md

Similar to `TWO_HEX_ATTACK_BREATH`, but affecting two additional hexes in a triangular formation from the target hex.
Deprecated. Please use [MULTIHEX_UNIT_ATTACK](#multihex_unit_attack) instead with custom icon and description.

### LONG_WEAPON

> The affected unit can perform a melee attack from two hexes away if there is exactly one empty hex between attacker and target (`attacker -> empty hex -> victim`).

Reviewer: From what I see unit will always pick long attack if possible - there is no way to actually pick which mode to use, right?

```suggestion
The affected unit will always perform a melee attack from two hexes away as long as there is exactly one empty hex between attacker and target (`attacker -> empty hex -> victim`).
```

Author: Done
